### PR TITLE
storage: add more details about Clevis

### PR DESCRIPTION
**Pull request by jlebon**
- Introduce Clevis pinning options better.
- Add a warning about TPM2 pinning alone and link to upstream docs.
- Add examples of Tang pinning and TPM2+Tang pinning.

### CODE REVIEW

File: modules/ROOT/pages/storage.adoc [hunk 1]
```diff
@@ -264,9 +264,13 @@ storage:
       with_mount_unit: true
 ----
 
-The root filesystem can also be moved to LUKS. In the case of the root filesystem the LUKS device must be backed by https://coreos.github.io/ignition/operator-notes/#clevis-based-devices[clevis]. There is simplified Butane config syntax for encrypting the root filesystem; for example:
+The root filesystem can also be moved to LUKS. In that case, the LUKS device must be pinned by https://coreos.github.io/ignition/operator-notes/#clevis-based-devices[Clevis]. There are two primary pin types available: TPM2 and Tang (or a combination of those using Shamir Secret Sharing).
 
-.Moving the root filesystem to LUKS
+WARNING: TPM2 pinning alone provides only limited protection. Make sure to understand its threat model before choosing between TPM2 and Tang pinning. For more information, see https://github.com/latchset/clevis/blob/master/src/pins/tpm2/clevis-encrypt-tpm2.1.adoc#threat-model[this section] of the Clevis TPM2 pin documentation.
```
Comment (cgwalters) [inline modules/ROOT/pages/storage.adoc:269]: How about:
`NOTICE: TPM2 pinning just binds the encryption to presence of the machine.`

Comment (jlebon) [reply modules/ROOT/pages/storage.adoc]: Thank! I tweaked it a bit from your suggestion, but kept it as a `WARNING` to increase the chances users read that box before continuing.

Comment (cgwalters) [reply modules/ROOT/pages/storage.adoc]: I'm OK to get this in as is.

But I also think `WARNING` is still too strong for this.  `IMPORTANT NOTICE:` ?

Comment (cgwalters) [reply modules/ROOT/pages/storage.adoc]: Or how about `SECURITY: `?

File: modules/ROOT/pages/storage.adoc [hunk 2]
```diff
@@ -264,9 +264,13 @@ storage:
       with_mount_unit: true
 ----
 
-The root filesystem can also be moved to LUKS. In the case of the root filesystem the LUKS device must be backed by https://coreos.github.io/ignition/operator-notes/#clevis-based-devices[clevis]. There is simplified Butane config syntax for encrypting the root filesystem; for example:
+The root filesystem can also be moved to LUKS. In that case, the LUKS device must be pinned by https://coreos.github.io/ignition/operator-notes/#clevis-based-devices[Clevis]. There are two primary pin types available: TPM2 and Tang (or a combination of those using Shamir Secret Sharing).
 
-.Moving the root filesystem to LUKS
+CAUTION: TPM2 pinning just binds encryption to the physical machine in use. Make sure to understand its threat model before choosing between TPM2 and Tang pinning. For more information, see https://github.com/latchset/clevis/blob/master/src/pins/tpm2/clevis-encrypt-tpm2.1.adoc#threat-model[this section] of the Clevis TPM2 pin documentation.
```
Comment (akostadinov) [inline modules/ROOT/pages/storage.adoc:269]: This link provides little information which treat models one is protected from and which - not. I guess I'm looking for a more comprehensive guide but well. It seems in general topic is complicated and I don't find well systematized information about it.

Comment (jlebon) [reply modules/ROOT/pages/storage.adoc:269]: Yes, I agree the upstream documentation could have more details about tradeoffs. I opened https://github.com/latchset/clevis/issues/381 to track this. If a section gets added, we can update these docs to link to it.

**Review by travier — APPROVED** [review]

**Pull request merged**